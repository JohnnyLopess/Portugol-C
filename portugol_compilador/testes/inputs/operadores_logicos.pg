programa
{
    funcao inicio()
    {
        // Teste do operador 'e'
        inteiro a = 10
        inteiro b = 20
        se (a > 5 e b < 25)
        {
            escreva("Condicao 1: Verdadeiro\n")
        }
        senao
        {
            escreva("Condicao 1: Falso\n")
        }

        // Teste do operador 'ou'
        inteiro x = 5
        inteiro y = 15
        se (x == 5 ou y == 10)
        {
            escreva("Condicao 2: Verdadeiro\n")
        }
        senao
        {
            escreva("Condicao 2: Falso\n")
        }

        // Teste do operador 'nao'
        logico estaAtivo = 1
        se (nao estaAtivo)
        {
            escreva("Condicao 3: Falso (inverteu)\n")
        }
        senao
        {
            escreva("Condicao 3: Verdadeiro (inverteu)\n")
        }

        // Combinação de operadores
        inteiro p = 1
        inteiro q = 2
        inteiro r = 3
        se ((p < q e q < r) ou nao (p == r))
        {
            escreva("Condicao 4: Verdadeiro\n")
        }
        senao
        {
            escreva("Condicao 4: Falso\n")
        }
        
        // Teste de propagação de constantes lógicas
        se (0 e 1) {
            escreva("Condicao 5: Nunca deveria aparecer\n")
        }
        
        // Teste de dead code elimination em while
        enquanto (0) {
            escreva("Condicao 6: Loop morto\n")
        }

        // Teste de dead code elimination em if
        se (0) {
            escreva("Condicao 7: If morto\n")
        } senao {
            escreva("Condicao 7: Else do if morto\n")
        }
    }
}