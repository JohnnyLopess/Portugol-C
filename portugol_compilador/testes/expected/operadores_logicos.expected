#include <stdio.h>
#include <string.h>

int main() {
    //Teste do operador 'e'
    int a = 10;
    int b = 20;
    if (a > 5 && b < 25) {
        printf("Condicao 1: Verdadeiro\n");
    } else {
        printf("Condicao 1: Falso\n");
    }
    //Teste do operador 'ou'
    int x = 5;
    int y = 15;
    if (x == 5 || y == 10) {
        printf("Condicao 2: Verdadeiro\n");
    } else {
        printf("Condicao 2: Falso\n");
    }
    //Teste do operador 'nao'
    bool estaAtivo = 1;
    if (!estaAtivo) {
        printf("Condicao 3: Falso (inverteu)\n");
    } else {
        printf("Condicao 3: Verdadeiro (inverteu)\n");
    }
    //Combinação de operadores
    int p = 1;
    int q = 2;
    int r = 3;
    if (p < q && q < r || !p == r) {
        printf("Condicao 4: Verdadeiro\n");
    } else {
        printf("Condicao 4: Falso\n");
    }
    //Teste de propagação de constantes lógicas
    //Teste de dead code elimination em while
    //Teste de dead code elimination em if
    printf("Condicao 7: Else do if morto\n");
    return 0;
}
